lhs.js

```diff
@@ -64,14 +64,7 @@ class Lhs {
      * when()
      */
     fnodesSatisfyingWhen(fnodes) {
-        const ret = [];
-        const pred = this._predicate;
-        fnodes.forEach(function (fnode) {
-            if (pred(fnode)) {
-                ret.push(fnode);
-            }
-        });
-        return ret;
+        return Array.from(fnodes).filter(this._predicate);
```

Comment:
I believe `fnodes` is always an array already, so you should be able to just call `filter` on it.

Reply:
Hm, so I tried that but it didn't work. When I printed fnodes, in one instance it was an array, but in the other it was a set.

Reply:
(Moving this back on-thread. Apparently, replies to GH emails no longer go to the right thread.) https://github.com/mozilla/fathom/blob/0b8153831d93456762357268b6e573dd66c128d7/lhs.js#L159 was the one I meant. It technically is correct, since [all I guarantee from fnodes()](https://github.com/mozilla/fathom/blob/0b8153831d93456762357268b6e573dd66c128d7/lhs.js#L54) is an iterable, but it's asking for trouble to be inconsistent.

Reply:
But whether you choose to correct that or not, all I guarantee is an iterable, so you would have to do Array.from or use a more general filter() procedure that isn't a method of Array. There's probably one in wu.

Reply:
Alright! So I should just leave this as is? Also, were you able to see when() in the docs?

Reply:
Since Erik is on PTO this week, I'll step in. This can probably be left as is.

I rendered the docs, and I see `when` in them, but without any details, just as a bare function reference. I'm not sure how to fix that. Can you render the docs locally, and see if you get the same thing?

Reply:
I actually hadn't written any details in, Erik told me to move them to a jsdoc comment, but maybe I interpreted incorrectly and I was supposed to leave them in the docs as well. I'm having a lot of trouble installing sphinx to render the docs. I can add the details in, but I'm not sure what to try now to get that fixed?

Reply:
Sphinx should be pulling the details from the jsdoc comments, so you shouldn't need anything in the .rst file. To render the docs, you can create a python virtualenv, and then install the requirements from [docs/tooling/doc-building-requirements.txt](https://github.com/mozilla/fathom/blob/master/docs/tooling/doc-building-requirements.txt) into it. Then you also need to `npm install jsdoc`. Finally, you can run `make html` in the doc directory.

If you have trouble with that, lets chat on IRC to see if we can get it sorted out.

Reply:
Ok there we go, the docs should be correct now